tool/internal.h

```diff
@@ -137,6 +137,11 @@ bool WriteToFile(const std::string &path, const uint8_t *in, size_t in_len);
 
 bool Ciphers(const std::vector<std::string> &args);
 bool Client(const std::vector<std::string> &args);
+// DoClient is a common function used to support the s_client option in both
+// bssl and openssl tools. It takes an additional parameter |tool| to indicate
+// which tool's s_client is being invoked. 1 indicates openssl and 0 indicates
+// bssl.
+bool DoClient(std::map<std::string, std::string> args_map, int tool);
```

Comment:
We can also make `tool` a boolean here. The `tool` naming also applies to both, so we should rename it to something like `is_openssl_s_client` , so it makes more sense when set to true or false.